content/client-lib-development-guide/features.textile

```diff
@@ -339,6 +340,14 @@ h3(#rest-encryption). Encryption
 h3(#rest-compatibility). Forwards compatibility
 * @(RSF1)@ The library must apply the "robustness principle":https://en.wikipedia.org/wiki/Robustness_principle in its processing of requests and responses with the Ably system. In particular, deserialization of Messages and related types, and associated enums, must be tolerant to unrecognised attributes or enum values. Such unrecognised values must be ignored.
 
+h3(#batch-operations). Batch Operations
+* @(BO1)@ The batch operations functions must use the REST endpoints in Batch Mode, sending a single request containing all specified data
+* @(BO2)@ Batch operations must be able to be performed for the following:
+** @(BO2a)@ @BatchOperations::publish@ publishes messages against one or more channels with one or more messages
+*** @(B02a1)@ Functions should be provided to pass either an array or a single object for both Message and Channel ID, or an array of @BatchSpec@ objects
```

Comment:
I would avoid 'array or a single object' as this language feels prescriptive and not programming language neutral (this could be called a slice in Go.)

Also is it worth mentioning that the BatchSpec object behaves like a matrix? In my mind I imagine it working like a matrix where all messages are applied to all channelIDs like...

```
[ 
    {
        channels: []channel{"1", "2"},
        messages: []messages{"1", "2"}
    },

    {
        channels: []channels{"3"},
        messages: []messages{"1"}
    }
]
```
This example batch spec would apply message "1" & "2" to both channels "1" & "2" and only apply message "1" to channel "3"

Reply:
> I would avoid 'array or a single object' as this language feels prescriptive and not programming language neutral (this could be called a slice in Go.)  
  
Ideally I think there would be a terminology guide for things like this. The point I wanted to get across with this wording vs something like 'one or more' is that, language permitting, it should be possible to pass either a single object or an array, as opposed to just an array with one or more objects.

Reply:
I was possibly thinking about this from a strongly typed language point of view. But the more I think about it, single object or multiple objects should be possible to get into a slice in Go with the `..` varadic operator.